[BREAKING] Implemented containsNestedMapping() (#9146)
ethereum/solidity: Changelog.md

 * JSON AST: Removes members with ``null`` value from JSON output.
 * Type Checker: Disallow shifts by signed types.
 * Type Checker: Exponentiation and shifts of literals by non-literals will always use ``uint256`` or ``int256`` as a type.

>  * Type Checker: Disallows structs and arrays that contains a nested mapping and lives in memory or calldata.

Reviewer: ```suggestion
 * Type Checker: Disallow structs and arrays in memory or calldata if they contain nested mappings.
```

>  * Type Checker: Disallows assignments to state variables that contains a nested mapping.

Reviewer: ```suggestion
 * Type Checker: Disallow assignments to state variables that contain nested mappings.
```

Language Features:
 * Yul: Disallow EVM instruction `pc()`.

Compiler Features:

>  * General: The function ``Type::canLiveOutsideStorage()`` is replaced by ``Type::containsNestedMapping()``.

Reviewer: This is an internal that should not be relevant to users.

 * JSON AST: Removes members with ``null`` value from JSON output.
 * Type Checker: Disallow shifts by signed types.
 * Type Checker: Exponentiation and shifts of literals by non-literals will always use ``uint256`` or ``int256`` as a type.
 * Type Checker: Disallow structs and arrays in memory or calldata if they contain nested mappings.

>  * Type Checker: Disallow assignments to state variables that contain nested mappings.

Reviewer 2: NTS: If state variable is a user defined type that contains a mapping, assignments to its non-mapping members is permitted. What is not permitted is constructing the said user defined type and assigning this to state variable.
